packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,448 @@
+      currentPill.current = rollUpPill.getAttribute('id');
+      setTabIndex('0');
+    }
+  }, [isCollapsed, isCollapsible]);
+
+  const handleFocusOnRollUpTrigger = useCallback(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    // To focus the immediate focusable pill after the rollUp pill is selected
+    if (isCollapsible && !isCollapsed && pills.length > 0) {
+      if (pills.length === React.Children.count(children)) {
+        setTabIndex('-1');
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+        if (isRollUpRemoved.current) { // if the rollup pill was triggered with keyboard focus the node
+          focusCurrentNode();
+          isRollUpRemoved.current = false;
+        }
+      }
+    }
+  }, [children, isCollapsed, isCollapsible]);
+
+  const handlePillDeletion = useCallback(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const startIndex = PillsUtils.handleRollUp(filterPillsRef);
+
+    if (isPillDeleted.current) {
+      if (React.Children.count(children) <= 0) {
+        focusPillsContainer();
+        return;
+      }
+      // if there is a roll Up pill, set tabindex to -1
+      if (rollUpPill) {
+        PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+      }
+
+      if (pills.length > 0) {
+        PillsUtils.setPillsTabIndex(pills, '-1');
+      }
+
+      if (pills.length === startIndex) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill && startIndex >= 1) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill) {
+        currentPill.current = rollUpPill.getAttribute('id');
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+      isPillDeleted.current = false;
+    }
+  }, [children]);
+
+  const handleResize = useCallback((entries) => {
+    if (!Array.isArray(entries)) {
+      return;
+    }
+
+    setUpdatedCount(React.Children.count(children));
+    setRollUpCount(React.Children.count(children));
+
+    if (isCollapsible) {
+      generateRollUp();
+    }
+
+    resetTabIndex();
+    handleFocusOnRollUpTrigger();
+    handlePillDeletion();
+  }, [children, isCollapsible, resetTabIndex, handleFocusOnRollUpTrigger, handlePillDeletion, generateRollUp]);
+
+  useLayoutEffect(() => {
+    let observer = new ResizeObserver((entries) => {
+      handleResize(entries);
+    });
+    observer.observe(filterPillsRef.current.parentNode);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, [children, handleResize]);
+
+  const focusNextNode = (pills, rollUpPill) => {
+    if (focusNode.current + 1 === pills.length || focusNode.current + 1 < pills.length) {
```

Comment:
Is this equivalent to `(focusNode.current + 1 <= pills.length)`?

Reply:
Fixed 51a9979